graphql/resolvers/user/mutations.js

```diff
@@ -3,10 +3,16 @@ import { UserInputError } from "apollo-server";
 
 import { authenticateToken, hash, verify } from "$/auth/auth.js";
 import jwt from "jsonwebtoken";
+import validator from "validator";
 
 export default {
     login: async (_, { username, password }) => {
-        verify_login_data(username, password);
+        try {
+            validate_password(password);
+            validate_username(username);
+        } catch (e) {
+            throw new UserInputError("Username or password incorrect.");
+        }
```

Comment:
Is it necessary to wrap these validators with try/catch block? This results in some loss of information about which part is incorrect (also, there is no such try/catch block in corresponding part of register mutation).